File: app/src/main/java/com/example/proyecto/LoginActivity.java
package com.example.proyecto;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.proyecto.Connection.ConnectionBD;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;

public class LoginActivity extends AppCompatActivity {

    EditText usuario, clave;
    TextView lblregistrar;
    Button btningresar;
    Connection con;

    // Constructor para inicializar la conexión
    public LoginActivity() {
        ConnectionBD instanceConnection = new ConnectionBD();
        con = instanceConnection.connect();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.login);

        // Referenciar los elementos del layout
        usuario = findViewById(R.id.txtUsuario);
        clave = findViewById(R.id.txtContraseña);
        lblregistrar = findViewById(R.id.lblRegistrar);
        btningresar = findViewById(R.id.btnIngresar);

        // Listener para el botón "Iniciar sesión"
        btningresar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new LoginActivity.login().execute();
            }
        });

        // Listener para el texto "Regístrate"
        lblregistrar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Iniciar la actividad de registro
                Intent reg = new Intent(getApplicationContext(), RegistrarActivity.class);
                startActivity(reg);
            }
        });
    }

    // Clase interna para manejar el inicio de sesión en segundo plano
    public class login extends AsyncTask<String, String, String> {
        String z = null;
        Boolean exito = false;

        @Override
        protected String doInBackground(String... strings) {
            if (con == null) {
                runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        Toast.makeText(LoginActivity.this, "Verifique su conexión", Toast.LENGTH_SHORT).show();
                    }
                });
                z = "Sin conexión";
            } else {
                try {
                    String sql = "SELECT * FROM USUARIO WHERE usuario = '" + usuario.getText() + "' AND CLAVE = '" + clave.getText() + "'";
                    Statement stm = con.createStatement();
                    ResultSet rs = stm.executeQuery(sql);

                    if (rs.next()) {
                        runOnUiThread(new Runnable() {
                            @Override
                            public void run() {
                                Toast.makeText(LoginActivity.this, "Acceso exitoso", Toast.LENGTH_SHORT).show();
                                Intent menu = new Intent(getApplicationContext(), MainActivity.class);
                                startActivity(menu);
                            }
                        });

                        usuario.setText("");
                        clave.setText("");
                    } else {
                        runOnUiThread(new Runnable() {
                            @Override
                            public void run() {
                                Toast.makeText(LoginActivity.this, "Error en el usuario o contraseña", Toast.LENGTH_SHORT).show();
                            }
                        });

                        usuario.setText("");
                        clave.setText("");
                    }

                } catch (Exception e) {
                    exito = false;
                    Log.e("Error de conexión", e.getMessage());
                }
            }
            return z;
        }

        @Override
        protected void onPreExecute() {
            super.onPreExecute();
        }

        @Override
        protected void onPostExecute(String s) {
            super.onPostExecute(s);
        }
    }
}

File: app/src/main/java/com/example/proyecto/LoginActivity.kt
package com.example.proyecto

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.DocumentSnapshot
import com.google.firebase.firestore.FirebaseFirestore

class LoginActivity : AppCompatActivity() {
    private lateinit var editTextCorreo: EditText
    private lateinit var editTextContrasena: EditText
    private lateinit var btnIniciarSesion: Button
    private lateinit var lblRegistrar: TextView

    private lateinit var mAuth: FirebaseAuth
    private lateinit var db: FirebaseFirestore

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.login)

        // Inicializar Firebase
        mAuth = FirebaseAuth.getInstance()
        db = FirebaseFirestore.getInstance()

        // Referencias a los campos del layout
        editTextCorreo = findViewById(R.id.txtUsuario)
        editTextContrasena = findViewById(R.id.txtContraseña)
        btnIniciarSesion = findViewById(R.id.btnIngresar)
        lblRegistrar = findViewById(R.id.lblRegistrar)

        // Acción para "Regístrate"
        lblRegistrar.setOnClickListener {
            val intent = Intent(this, RegistrarActivity::class.java)
            startActivity(intent)
        }

        // Acción del botón "Iniciar sesión"
        btnIniciarSesion.setOnClickListener {
            val correo = editTextCorreo.text.toString()
            val contrasena = editTextContrasena.text.toString()

            // Validación de campos vacíos
            if (correo.isEmpty() || contrasena.isEmpty()) {
                Toast.makeText(this, "Por favor, completa los campos.", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            // Autenticar usuario con Firebase Authentication
            mAuth.signInWithEmailAndPassword(correo, contrasena)
                .addOnCompleteListener { task ->
                    if (task.isSuccessful) {
                        val userId = mAuth.currentUser?.uid
                        if (userId != null) {
                            // Obtener datos del usuario desde Firestore
                            db.collection("usuarios").document(userId).get()
                                .addOnSuccessListener { document: DocumentSnapshot ->
                                    if (document.exists()) {
                                        val nombre = document.getString("nombre")
                                        val intent = Intent(this, BienvenidaActivity::class.java)
                                        intent.putExtra("nombre", nombre)
                                        startActivity(intent)
                                        finish()
                                    } else {
                                        Toast.makeText(
                                            this,
                                            "Usuario no encontrado en Firestore.",
                                            Toast.LENGTH_SHORT
                                        ).show()
                                    }
                                }
                                .addOnFailureListener { e ->
                                    Toast.makeText(
                                        this,
                                        "Error al obtener datos: ${e.message}",
                                        Toast.LENGTH_SHORT
                                    ).show()
                                }
                        } else {
                            Toast.makeText(
                                this,
                                "Error: no se pudo obtener el ID del usuario.",
                                Toast.LENGTH_SHORT
                            ).show()
                        }
                    } else {
                        Toast.makeText(
                            this,
                            "Error al iniciar sesión: ${task.exception?.message}",
                            Toast.LENGTH_SHORT
                        ).show()
                    }
                }
        }
    }
}
